aws/rust-runtime/aws-config/src/imds/client.rs

```diff
@@ -244,7 +235,7 @@ struct ImdsCommonRuntimePlugin {
 
 impl ImdsCommonRuntimePlugin {
     fn new(
-        connector: DynConnector,
+        http_client: Option<SharedHttpClient>,
```

Comment:
The default http client runtime plugin is registered by the `Operation` orchestrator utility, so if this is set to None, it will use the default HttpClient implementation. `aws-config` is no longer responsible for setting a default http client.